src/test/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryTest.java

```diff
@@ -899,7 +900,36 @@ public void testAFTS()
         aftsQueryWithCount("=ASPECT:\"test:testSuperAspect\" AND =test:orderMLText:\""+stringValue+"\"", 1);
         
     }
+
+    @Test
+    public void testAftsPagination()
+    {
+        String query = "=TYPE:\"cm:folder\" ";
+        int numFolders = 6;
+        int count = 4;
+        ResultSet results = query(SearchService.LANGUAGE_FTS_ALFRESCO, query, count);
+        assert(results.length() == count);
+        assert(results.getNumberFound() == numFolders);
+    }
     
+    public ResultSet query(String ql, String query, Integer limit)

```

Comment:
It would be worth adding some Javadoc here too, otherwise no one will be able to reuse this method.